content/docs/hoon/guides/sail.md

```diff
@@ -0,0 +1,495 @@
++++
+title = "Sail (HTML)"
+weight = 6
+template = "doc.html"
++++
+
+Sail is a domain-specific language for composing HTML structures in Hoon.
+Front-ends for Urbit apps are often created and uploaded separately to the rest
+of the code in the desk. Sail provides an alternative approach, where front-ends
+can be composed directly inside agents.
```

Comment:
"Like everything else in Hoon, a Sail document is a binary tree data structure, just one produced by a specialized markup language within Hoon."